### PR TITLE
doc(gradle): Adding Minikube getting started

### PR DESCRIPTION
**Pull request by sunix**
## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have implemented unit tests to cover my changes
 - [x] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
+[[quarkus-scenario]]
+== Quarkus
+JKube provides ready to use generators. A JKube generator is composed of default configurations to build container images, generate k8s resources and deploy them to kubernetes clusters. There are predifined generators for each type of Java project. For instance, we have a generator for Quarkus projects. It helps JKube to identify that a project is a Quarkus one and provides the right configuration to be used.
```
Comment (rohanKanojia) [inline gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc:3]: > A JKube generator is composed of default configurations to build container images, generate k8s resources and deploy them to kubernetes clusters.


I think maybe we should change this line to "A JKube Generator is composed of opinionated image configurations for building container images, this image configuration is later used by enrichers to generate Kubernetes resources"

**Comment by rohanKanojia** [comment]
Shall we also include Quarkus native build in this scenario?

File: gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc (hunk 1 above)
Comment (sunix) [reply gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc:3]: OK Somehow I miss understood the scope of generator, I will provide another proposal for this part

**Comment by sunix** [comment]
> Shall we also include Quarkus native build in this scenario?

maybe, how do you turn on the native build ? If this is simple, I can include that in it.

**Comment by rohanKanojia** [comment]
> maybe, how do you turn on the native build ? If this is simple, I can include that in it.

Need to check, In maven it could be turned on by providing some properties https://github.com/eclipse/jkube/blob/master/quickstarts/maven/quarkus/pom.xml#L110

**Comment by rohanKanojia** [comment]
I'm wondering whether we should keep quarkus scenario in root of scenario section or create a new section "getting started" and place it as a child. Something like this:
Screenshot from 2021-12-16 12-06-51

What do you think about this? Otherwise, I'm okay with the current position as well.

File: gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+[[quarkus-scenario-zero-configuration]]
+=== Zero configuration
+1. Start by generating a new Gradle Quarkus project from https://code.quarkus.io. Make sure Gradle is selected. Download the generated project and extract the content.
+
+2. Open the `build.gradle` file and add the plugin in the `plugins` section.
+
+   plugins {
+     id 'java'
+     id 'io.quarkus'
+     id 'org.eclipse.jkube.kubernetes' version '1.6.0-SNAPSHOT'
```
Comment (rohanKanojia) [inline gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc:29]:
```suggestion
     id 'org.eclipse.jkube.{pluginExtension}' version '{version}'
```

File: gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+== Quarkus
+JKube provides ready to use generators. A JKube generator is composed of default configurations to build container images, generate k8s resources and deploy them to kubernetes clusters. There are predifined generators for each type of Java project. For instance, we have a generator for Quarkus projects. It helps JKube to identify that a project is a Quarkus one and provides the right configuration to be used.
+
+This is an example of how you can use the JKube zero configuration for a Quarkus project.
+
+[[quarkus-scenario-prerequisites]]
+=== Prerequisites
+You will need the following for the scenario:
+
+* `minikube` installed and running on your computer
```
Comment (rohanKanojia) [inline gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc:11]: I just realized this same documentation would be used for OpenShift Gradle Plugin. Do you think we should show this only for Kubernetes Gradle Plugin?

```adoc
ifeval::["{task-prefix}" == "k8s"]

// scenario on minikube 

endif::[]

ifeval::["{task-prefix}" == "oc"]

// scenario on CRC/OpenShift Cluster

endif::[]
```

File: gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc (hunk 2 above)
Comment (sunix) [reply gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc:32]: Selection_579
it doens't seem to work

Comment (sunix) [reply gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc:32]: ok found how to fix it

Comment (sunix) [reply gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc]: fixed

File: gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc [hunk 4]
```diff
@@ -0,0 +1,116 @@
+[[quarkus-scenario]]
+== Get started with Quarkus and Minikube
+You can easily get started with using {plugin} on a https://quarkus.io/[Quarkus] without providing any explicit configuration. {plugin} would generate an opinionated container image and manifests by inspecting your project configuration.
```
Comment (manusa) [inline gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_quarkus.adoc:3]: `You can easily get started with using {plugin} [...]`

I think there might be some word missing in between with and using.

Or maybe just remove using.

**Comment by manusa** [comment]
As discussed internally:
  - @sunix `yeah but i think it is good to have something specific to get started.`
  - @manusa 
    - Having a specific Minikube section (How to share docker daemon, How to create an Ingress, etc.)
    - Have full User workflows for each Framework including every main task (build, push, resource, apply, undeploy)
    - Minikube specific skippable tasks for each framework can have a small hint that points to the Minikube specific FAQ.
  - Let's try to add different get started scenarios and then refactor or group common parts

This PR is aligned with the current agreement.

However, this PR provides documentation which is invalid for OpenShift.
`gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_index.adoc` should contain a flag check to omit the `_quarkus.adoc` doc fragment in case of OpenShift.

**Comment by manusa** [comment]
As agreed during today's call, let's structure (for now) our documentation like:

* Getting Started
  * Minikube (visible only for Kubernetes)
    * Basic workflow
    * How to…
  * TODO - Free vanilla Kubernetes cluster (Outer-loop related)
  * Dev Sandbox (Visible only for OpenShift)
    * Basic workflow
    * How to disable routes?
  * Spring Boot
    * How to add a liveness and readiness probe?
  * Vert.x
    * How to set the (service) port
  * Quarkus
    * Zero configuration?? Final point GOTO $cluster
    * Native???
    * How to add a liveness and readiness probe?
  * Dockerfile

**Comment by sunix** [comment]
I have published the generated doc for this branch here: https://jkubedoc-sunix-jkube-minikube-getting-started.surge.sh/

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**